src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -142,12 +140,12 @@ public void NonRepeatingTimer_ThatHasAlreadyFired_CanChangeAndFireAgain()
     {
         AutoResetEvent are = new AutoResetEvent(false);
         TimerCallback tc = new TimerCallback((object o) => are.Set());
-        using (var t = new Timer(tc, null, 1, Timeout.Infinite))
+        using (var t = new Timer(tc, null, 10, Timeout.Infinite))
         {
-            Assert.True(are.WaitOne(50), "Should have received first timer event");
-            Assert.False(are.WaitOne(20), "Should not have received a second timer event");
+            Assert.True(are.WaitOne(250), "Should have received first timer event");
+            Assert.False(are.WaitOne(100), "Should not have received a second timer event");
```

Comment:
These Assert.False(are.WaitOne(...)) cases are the ones we need to be more stringent with, because we're waiting for something that we expect to never happen, so in the success case it's going to take the full amount of time we specify.  100ms may still be too small though for all of the fluctations possible in the CI environment.  I'd suggest going to at least 250.